### ISSUE TITLE
Downloading books from Internet Archive sets default filename to "cors_get.php"

### ISSUE THREAD

**Issue by MomoTwo** [johnfactotum/foliate#1385]
**Describe the bug**
Downloading free books from Internet Archive 
opens the save file dialog with file name: "cors_get.php"
instead of for example: "imperialism-and-world-economy.epub".

Saving the file with the selected file extension 
opens the document without issues. 

**To Reproduce**
1. Go to Catalogs: "Internet Archive".
2. Click on a book with the tag: "free".
3. Click on download or download as.
4. File Dialog opens and displays Name:"cors_get.php" 

**Expected behavior**
When the file dialog opens it should display the title of the book 
or a different download title as the file name 
and the default or selected file format as file extension.

**Version:**
 - Device:                                    SteamDeck
 - Foliate version:                        3.1.1
 - OS/Distribution and version:  SteamOS     3.5.19
 - Desktop environment:            KDE Plasma 5.27.5
 - KDE Framework version:         5.107.0
 - Qt version:                              5.15.9
 - Installation method:               Flatpak (Discover 5.27.5)

**Additional context**
This only occurs in the Internet Archive catalog
and might be just an issue with Internet Archive itself.
The epub version of the books looks weird sometimes 
not sure if this is normal. (Looks like an encoding issue.)

**Comment by johnfactotum** [comment]
> might be just an issue with Internet Archive itself

Yes. The suggested filename is provided by WebKit. It probably comes from the `Content-Disposition` header.

> The epub version of the books looks weird sometimes
> not sure if this is normal. (Looks like an encoding issue.)

It's probably normal. Most of their EPUBs are produced using OCR.
